### PR TITLE
Chronos : Support val data in autoformer fit

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

We need to align function in BaseForecaster and AutoformerForecaster synchronously.

### 1. Why the change?

https://github.com/intel-analytics/BigDL/issues/5834

### 2. User API changes

```
from bigdl.chronos.forecaster import AutoformerForecaster
forecaster = AutoformerForecaster(...)
forecaster.fit(data, validation_data, validation_mode='output', earlystop_patience=1)
```

### 3. Summary of the change 

- [x] support validation_data, validation_mode and earlystop_patience for AutoformerForecaster.fit
- [x] support get_model for AutoformerForecaster
- [x] code refactor

### 4. How to test?
- [ ] Unit test
- [ ] Application test

### CODE REVIEW

File: python/chronos/src/bigdl/chronos/forecaster/autoformer_forecaster.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import numpy as np
 from pandas import Timedelta
 from bigdl.chronos.forecaster.abstract import Forecaster
+from bigdl.chronos.forecaster.utils import *
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/autoformer_forecaster.py:21]: please use `from bigdl.chronos.forecaster.utils import xxx, xxx, xxx`

File: python/chronos/test/bigdl/chronos/forecaster/test_autoformer_forecaster.py [hunk 2]
```diff
@@ -270,8 +271,7 @@ def test_autoformer_forecaster_confidence_interval_with_loader(self):
                                           output_feature_num=2,
                                           label_len=12,
                                           freq='s',
-                                          seed=0,
-                                          moving_avg=20) # even
+                                          seed=0)
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/forecaster/test_autoformer_forecaster.py:274]: why are we removing this even moving_avg test?

Comment (rnwang04) [reply python/chronos/test/bigdl/chronos/forecaster/test_autoformer_forecaster.py:274]: even moving_avg is test in test_autoformer_forecaster_even_kernel. So this test is probably not needed here.

File: python/chronos/src/bigdl/chronos/forecaster/autoformer_forecaster.py (hunk 1 above)
Comment (rnwang04) [reply python/chronos/src/bigdl/chronos/forecaster/autoformer_forecaster.py]: done

**Review by TheaperDeng — APPROVED** [review]
LGTM

Some comment: we might need an issue states we need to remove the magic number/magic string in our code. we may use a utils file such as format.py in future to store all the default paths and some thing

**Pull request merged**